tobac/tests/test_feature_detection.py

```diff
@@ -54,10 +54,49 @@ def test_feature_detection_multithreshold_timestep(
     )
 
     # Make sure we have only one feature
-    assert len(fd_output.index) == 1
+    assert len(fd_output) == 1, f"Expected 1 feature, but got {len(fd_output)}"
     # Make sure that the location of the feature is correct
-    assert fd_output.iloc[0]["hdim_1"] == pytest.approx(test_hdim_1_pt)
-    assert fd_output.iloc[0]["hdim_2"] == pytest.approx(test_hdim_2_pt)
+    assert fd_output.iloc[0]["hdim_1"] == pytest.approx(
+        test_hdim_1_pt
+    ), f"Expected hdim_1 to be {test_hdim_1_pt}, but got {fd_output.iloc[0]['hdim_1']}"
+    assert fd_output.iloc[0]["hdim_2"] == pytest.approx(
+        test_hdim_2_pt
+    ), f"Expected hdim_2 to be {test_hdim_2_pt}, but got {fd_output.iloc[0]['hdim_2']}"
+
+    labels, features = feat_detect.feature_detection_multithreshold_timestep(
+        test_data_xr,
+        0,
+        threshold=test_threshs,
+        n_min_threshold=n_min_threshold,
+        dxy=dxy,
+        wavelength_filtering=wavelength_filtering,
+        return_labels=True,
+    )
+
+    # Make sure we have only one feature
+    assert (
+        len(features.index) == 1
+    ), f"Expected 1 feature, but got {len(features.index)}"
+
+    # Check if labels are returned
+    assert isinstance(
+        labels, xr.DataArray
+    ), "Expected label fields to be a xarray.DataArray"
+
+    # Check if labels have the correct shape
+    assert labels.shape == (
+        test_data_xr.shape[0],
+        test_data_xr.shape[1],
+    ), f"Expected labels shape to be {test_data_xr.shape}, but got {labels.shape}"
+
```

Comment:
Can we add a more exact test for the label returns here. E.g. that the only nonzero values in label_field match the feature value in the returned dataframe, and that all of these points are >= the threshold value in the input field

Reply:
Good point! We will contribute more tests!